quantum/rgblight.c

```diff
@@ -620,7 +623,11 @@ void rgblight_set(void) {
 #    else
     start_led = led + clipping_start_pos;
 #    endif
-    ws2812_setleds(start_led, num_leds);
+
+#ifdef RGBW
+    convert_rgb_to_rgbw(start_led);
```

Comment:
Does this not need to run for `num_leds`, at the moment it would just convert the first one?

Reply:
Hmm, it probably does.  Thanks!

Reply:
while a little inefficient if also using RGBLIGHT_LED_MAP, it would be something like:
```suggestion
    for (uint8_t i = 0; i < num_leds; i++) {
        convert_rgb_to_rgbw(start_led[i]);
    }
```

Reply:
Yeah, I was thinking about that, but that inefficiency stopped me.

Reply:
The thing is working out which is going to be the hot code pathways, as you have it now is objectively worse for performance.
```c
    for (uint8_t i = effect_start_pos; i < effect_end_pos; i++) {
        if (!rgblight_config.enable) {
```
When using non RGBW leds, which i would says is the more common use case, this will now enumerate the whole list, no matter enabled status. For something like 10 LEDs, the code block always does 10 loop iterations, and 10 if checks, vs before when it was 1 if check and maybe 10 for loop iterations.